me first a pull request

spokik opened the pull request.

spokik committed:
last work version from my job

spokik committed:
Пересобрал в виде расширения

В дальнейшем планирую разделить это все на модули.
как вариант изучить webpac. Сейчас некоторые браузеры не поддерживают часть функций, (у одной девушки crome v49..._

spokik committed:
Частично реализованны модули

spokik committed:
Перенес авто Мд в модуль

spokik committed:
modules.js модули
собираются через функцию

spokik committed:
Все разнесено по модулям

spokik committed:
rename and correct mistakes in selectYear

spokik committed:
Исправлена табуляция на единую в 2 пробела

spokik committed:
Не добавляет кучу составов при мутации элемента

spokik committed:
Merge branch 'spokik-dev-events' into spokik-as-extension

spokik committed:
asd

CristinaVolk commented on kad.js:
```diff
@@ -5,9 +5,6 @@ import { windowForUserScript } from './modules/alert.js';
 import { defaultMD } from './modules/autoMD.js';
 
 if (document.location.href === `https://kad.arbitr.ru/`) { windowForUserScript() }
```
strings that contain url are better to be stored into const

CristinaVolk commented on kad.js:
```diff
@@ -5,9 +5,6 @@ import { windowForUserScript } from './modules/alert.js';
-
 
 if (document.location.href != `https://kad.arbitr.ru/`) {
   let usersSettings = JSON.parse(localStorage.getItem(`usersSettings`))
```
try to keep one pattern of quotes, single or double or backward. Usually, if you have just a string use the single quotes `''`

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+    footer.innerHTML = `${xc} <div id="activeStat" class="tampleteButton" style="height:23px">Статистика</div>`;
+
+    let classBlok = document.createElement("style")
+    classBlok.innerHTML = "div.tampleteButton{height: 30px;width: 130px;-webkit-appearance: button;-webkit-writing-mode: horizontal-tb !important;color: buttontext;text-shadow: none;display: inline-block;text-align: center;align-items: flex-start;background-color: buttonface;box-sizing: border-box;margin: 0em;font: 400 13.3333px Arial;padding: 1px 6px;border-width: 2px;border-style: outset;border ...
```
you can separate html from css. Create a module css and [ut all the styling there ;) when you use everything together it is not possible to change in future when the project grows and it looks dirty, not readable.

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+    for (let index = 0; index < sostavs.length; index++) {
+        let starArray = JSON.parse(localStorage.getItem(`sostav${sostavs[index]}`))
+        for (let key in starArray) {
+            if (starArray[key].month == month) {
```
try not to compare with  `==`, use triple one ;)

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+    let today = new Date().getDate()
+    let month = new Date().getMonth() + 1
+    let counter = {}
+    for (let index = 0; index < sostavs.length; index++) {
```
I believe you can replace this loop with the `map` method :)

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+    //string stringify
+    let stringReturn = ''
```
try to use more descriptive and meaningful names. It's not clear what you mean by stringReturn, which string? what return? why xd?

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+    let stringReturn = ''
+    for (let key in counter) {
+
+        stringReturn = stringReturn + `${key} состав: ${counter[key]} штук | `
```
please check if you can replace this loop just by that line:
`const stringReturn = counter.map((key, index) => (`${index} состав: ${key} штук | `));`

spokik replied:
Угу, вот только counter = {}

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+    return stringReturn
+}
+
+export { addBattons }
```
check please the spelling of the filename ;)

CristinaVolk commented on modules/addBattons.js:
```diff
@@ -0,0 +1,55 @@
+    let xc = sostavPerDay(usersSettings.usersSS)
+    footer.innerHTML = `${xc} <div id="activeStat" class="tampleteButton" style="height:23px">Статистика</div>`;
+
+    let classBlok = document.createElement("style")
```
`classBlok` looks weird tbh , doesn't it?

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+    let promise = new Promise((resolve, reject) => {
+        let observer = new MutationObserver((mutationRecords) => {
+            if (mutationRecords.length > 1) {
+                let saveScanStat = document.querySelector("#b-container > div.b-popup-wrapper.js-popup-wrapper.js-popup-wrapper--upload > div.b-popup.b-popup--blue.b-popup--upload.js-popup--upload.b-popup--edit.js-popup--edit > form > div.b-popup-button.js-upload-submit"); // кнопка по которой срабатывает скрипт
```
please try mot to shorten the names, not to be afraid of using long names. They are descriptive and takes less time to understand what's going on :)

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+function addStaticOnLocalStorage() {
+    let elem = document.querySelector("#b-container > div.b-popup-wrapper.js-popup-wrapper.js-popup-wrapper--upload > div.b-popup.b-popup--blue.b-popup--upload.js-popup--upload")
+
+    let promise = new Promise((resolve, reject) => {
```
if you don't use the parameter `reject` , then you can add underscore before in order to avoid eslint error/warn .
And if you do not change the variable in code use const instead

CristinaVolk commented on README.md:
```diff
@@ -1,17 +1,17 @@
 # Kad.arbitr
+
```
Here should be instructions how to run your application in Readme

CristinaVolk commented on Scan:
```diff
@@ -1,268 +1,520 @@
-        document.getElementsByClassName(`b-popup-sj-link js-popup-sj-link js-popup-sj-link--upload`)[0].onclick = () =>{document.getElementsByClassName('js-input js-input--combobox js-input--deselect_with_save_val')[1].value = `Материалы по делу`;}
+(function () {
+  "use strict";
+  let usersSettings = JSON.parse(localStorage.getItem(`usersSettings`))
```
use const if you don't assign a new value ;)

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -88,7 +90,7 @@ function addStaticOnLocalStorage() {
+    elem.innerHTML = `<div class="b-popup-info-title" id="userSS">Состав</div><span class="b-popup-info-text js-popup-info-text" title="${sostav}">${sostav}</span>`
     let isYourSostav = false
     function isCorrectSostav() {
       for (let i = 0; i < usersSettings.usersSS.length; i++) {
```
the content of this loop can be written just in one line with the help of the ternary operation:
`for (let i = 0; i < usersSettings.usersSS.length; i++) {
        return (isYourSostav =
          usersSettings.usersSS[i] == sostav ? true : false);
      }`

CristinaVolk replied:
and also below ` состав неопределен` is it should be necessary the phrase with space before??

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+
+            //Срабатывает прии НАЖАТИИ на "Сохранить"
+            params.saveScanStat.onclick = () => {
+                let now = new Date()
```
please change the name to `currentDate` for example :)

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+                let now = new Date()
+                let dataForLS = {}
+                dataForLS = JSON.parse(localStorage.getItem(`sostav${params.sostav}`))
+                if (dataForLS == null) {
```
you can reverse this logic of checking :)
check for existence ;)
`const dataForLS = JSON.parse(localStorage.getItem(`sostav${params.sostav}`));`
and then `if (dataForLS !==null)`

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+        let observer = new MutationObserver((mutationRecords) => {
+            if (mutationRecords.length > 1) {
+                let saveScanStat = document.querySelector("#b-container > div.b-popup-wrapper.js-popup-wrapper.js-popup-wrapper--upload > div.b-popup.b-popup--blue.b-popup--upload.js-popup--upload.b-popup--edit.js-popup--edit > form > div.b-popup-button.js-upload-submit"); // кнопка по которой срабатывает скрипт
+                let q = document.querySelector("#chrono_list_content > div.b-chrono-items-container.js-chrono-items-container > div").childElementCount  //Кол-во элементов для фикла
```
you can use such short names for large functions or classes, where another context gives you the entire impression, but here it's tricky to understand

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+                    saveScanStat: saveScanStat,
+                    sostav: sostav,
+                    docId: docId,
+                    a40: a40,
```
if you use the same names of the props and values , then you don't need to write it twice ;)
` let params = {
                    saveScanStat,
                    sostav,
                    docId,
                    a40,`

CristinaVolk replied:
`        let params = {
          ...params,
          saveScanStat,
          sostav,
          docId,
          a40,
        };`

spokik replied:
Я собственно так и сделал недавно, правда в другой ветке

CristinaVolk commented on modules/addStaticOnLocalStorage.js:
```diff
@@ -0,0 +1,117 @@
+                            .trim() //Удаляет пробелы "с краёв"
+                            .split(` `) //Правращает в масив по разделителю " "
+                        let stringValueCounter = 0
+                        if (stringValueCounter < 2) {
```
and what if `stringValueCounter` is equal to 2 ?
I recommend you to change if-else implementation here, it's confusing and long

CristinaVolk commented on modules/addTamplaties.js:
```diff
@@ -1,57 +1,57 @@
 // Добавляет обработчики интерфейса
 function addTamplaties() {
```
the name :)

CristinaVolk commented on modules/addTamplaties.js:
```diff
@@ -0,0 +1,57 @@
+// Добавляет обработчики интерфейса
```
you should apply VS formatter here. It's unreadable at all

CristinaVolk replied:
![Uploading image.png…]()

spokik committed:
оп, Хлоп

CristinaVolk commented on modules/checkReload.js:
```diff
@@ -0,0 +1,41 @@
+        if (lastFilles === undefined) {
+            lastFilles = {}
+            return lastFilles
+        }
```
why do you assign an empty object?
you can check the value to exist and proceed

spokik replied:
Я уже не помню, оно без этого не работало XD

CristinaVolk commented on modules/statisticsInterface.js:
```diff
@@ -0,0 +1,237 @@
+    input:checked[type="checkbox"]::before {
+      left: 20px;
+    }
+    `
```
you need to extract all the css to the css module :)

spokik replied:
ok. I will be try

CristinaVolk commented on modules/alert.js:
```diff
@@ -0,0 +1,64 @@
+        if (year20 == true) { year = 20; }
+        else if (year19 == true) { year = 19; }
+        else if (year18 == true) { year = 18; }
+        else { alert(`ЕРОР! Чет какая - то шляпа введина в место года`) }
```
you can use ternary operators here

CristinaVolk commented on modules/alert.js:
```diff
@@ -0,0 +1,64 @@
+    //пересобирает строку, заменя 2 последний цифры
+    function setYear(year) {
+        let newNumber = document.querySelector(`#sug - cases.tag input[type = "text"]: valid`).value
+        let inString = document.querySelector(`#sug - cases.tag input[type = "text"]: valid`).value["length"]
```
i don't like the name XD

CristinaVolk commented on modules/alert.js:
```diff
@@ -0,0 +1,64 @@
+    }
+
+
+    //Красим фон и меняем год
```
if you use comments, it means you use bad names ;)

CristinaVolk commented on modules/statisticsInterface.js:
```diff
@@ -138,98 +138,98 @@ function statisticsInterface() {
-            localStorage.setItem(`usersSettings`, JSON.stringify(preSet))
-        })
+      }
+      preSet.usersSS = []
```
too long, i believe you can reduce it with the help of methods of js and ternary operators ;)

CristinaVolk commented on modules/usersSettings.js:
```diff
@@ -1,13 +1,13 @@
 const usersSettings = JSON.parse(localStorage.getItem(`usersSettings`))
 if (usersSettings == undefined) {
```
`if (!usersSettings)`

spokik committed:
тык

spokik committed:
Добавилась генирация конок выбора состава, в случае, если состав не определен. не тестилось

spokik committed:
readme по бырому

spokik committed:
йцу

spokik committed:
Восстанавливаю из рабочей версии

Давно не занимался проектом, не помню где ошибки. Не хочу разбираться. Кажется более простым вариантом вернуться к рабочей версии и плясать от нее

spokik committed:
Восстанавливаю из рабочей версии
